### DISCUSSION TITLE
Evaluación de la E0

### DISCUSSION THREAD

**Discussion by ypne** [IIC2413/Syllabus-2024-2#71] [General]
El objetivo de la E0 es que aprendan a programar en PHP y para ello se les entrega un problema de limpieza de datos. Lo que se evaluará es:
1.- Correctitud del programa
     -Que el programa les los datos del input (archivos) y cargue las matrices
     -Que las consultas entreguen el resultado esperado para los datos de entrada (y sus correcciones)
2.- Limpieza de datos
     - Que el programa detecte las violaciones de formato (int, bool, string, lista de valores) y los escriba en un archivo aparte
     - Que el estudiante entregue una solución a los casos de violación de formato, ya sea corrigiendo, marcando como dato fuera de formato, o borrando el dato que produce la violación y todas las tuplas asociadas a él.

**Comment by cataruze** [comment]
Hola, es necesario tener un archivo aparte que contenga las violaciones de formato? Dato que esto no estaba en el enunciado o se refiere a escribir en el informe?

**Reply by vicntealonso** [reply]
Me sumo a la duda

**Reply by ypne** [reply]
No es necesario, pero es conveniente. Si no registras las violaciones, ¿cómo las vas a revisar y corregir después? Las puedes imprimir en pantalla también o puedes decirle al programa que se detenga en cada error y revisar+corregir de a una, pero puede ser más lento.
La tarea es tomar un input procesarlo automáticamente y detectar violaciones a las reglas; luego corregirlas y ejecutar las consultas solo con datos válidos
